### PR TITLE
Merge test

### PR DESCRIPTION
**Pull request by jmhernan**
@alastairmatheson, you've done a lot of great work with these massive and messy files and documented things well, thank you for that!  You will see that I tried not to break your code too much. We can align things first and then we can work together on making things more efficient and testable. Here are some of the major additions/changes:
# Metadata file integration
In the beginning of each script you will see there are these additional lines of code added:
```
library(RJSONIO)
library(RCurl)

script <- RCurl::getURL("https://raw.githubusercontent.com/jmhernan/Housing/uw_test/processing/metadata/set_data_env.r")
eval(parse(text = script))

METADATA = RJSONIO::fromJSON("//home/ubuntu/data/metadata/metadata.json")

set_data_envr(METADATA,"kcha_data")
```
I created a new directory `metadata` inside `processing` which contains two things:
1. `metadata_template.json`
   - This is the template to use to populate your corresponding directory paths and file names.  The .json file looks like this:
  ```
  {
    "sha_data":{
        "sql": "FALSE",
        "UW": "TRUE",
        "sha_path": "//path_to/SHA_data_folder/",
        "sha3a_new_fn": "file_name.type",
        "sha3b_new_fn": "file_name.type",
        .
        .
        .
  ```
  the `"sql"` and  `"UW"` are used to set up conditionals on chunks of code that correspond to say sql calls.  The `UW` condition I hope is temporary, there are some large chunks of code that do major data munging and cleaning that call specific column names that do not exist in the current UW data, so in an attempt to not totally break your code because of non-matching columns I copied a version that will work on our data for now and left your version unchanged. You will see something like:
  ```
  if (UW == TRUE) {
      #RUN THIS CODE THAT I CHANGED
  } else {
      #RUN ORIGINAL CODE
  }
  ```
  My hope is that when we integrate the new data we will have matching columns? (as of this PR I only incorporated the new KCHA data and waiting on SHA). 

  ### How to set-up:
  - Rename the `metadata_template.json` to `metadata.json` (it is already in the .gitignore) and fill in your corresponding parameters.  

2. `set_data_env.r`
   - This function parses the .json and assigns the variable parameters.
   - It takes 2 inputs, `metadata` and which data you're working on (3 possible inputs `sha_data`, `kcha_data`, and `combined`)

# Notable changes 
## 02_sha_sql_load.R 
This script had the most diff in terms of what worked for the current data that we have.  There were some fixes that were incorporated in the last UW merge that were missing in the new file.  There were some additional files that were previously used that I had to re-incorporate (i.e. `sha_vouch_type`). In addition to that there was a conflict with the variable mappings csv.  I attempted to add some missing variable mappings (which I did and seemed to fix things, but ultimately had to use the old mappings to work with the `sha_vouch_type` data (lines 197-209). We might need to work on this one together.  

## 06_pha_address_cleaning.R and 06a_pha_geocoding.R
You mentioned that this step could be moved to the end, I totally agree.  For now, I used geocoded files given to us `ESRI` and `google` to complete this step without actually running the geocoding part.  When we get new data it will need to be geocoded and I'm thinking we should set up a way to incorporate new data without running the whole pipeline again. 

I'm totally open to your input as well, feel free to amend things.

### CODE REVIEW

File: processing/02_sha_sql_load.R [hunk 1]
```diff
@@ -199,11 +246,12 @@ inc_clean_f <- function(df) {
   ### Summarize income and assets differently depending on data format
   # Tested out summarise instead of mutate in first part. No faster.
   # Still need ways to optimize this code
+  # Removed 'increment' not in UW data
```
Comment (alastairmatheson) [inline processing/02_sha_sql_load.R:250]: We should check why this is and make sure we're working from the same data

**Review by alastairmatheson — CHANGES_REQUESTED** [review]
Just need to create a separate inc_clean_f function for UW data in 02_sha_sql_load.R

File: processing/02_sha_sql_load.R (hunk 1 above)
Comment (jmhernan) [reply processing/02_sha_sql_load.R:250]: Have included and will do a revised PR.

**Comment by jmhernan** [comment]
@alastairmatheson, I've patched the income function. Let me know if there are any others I missed.

**Review by alastairmatheson — APPROVED** [review]

**Pull request merged**